### DISCUSSION TITLE
Differences Between MESSAGE-V and MESSAGE-ix

### DISCUSSION THREAD

**Discussion by zsalimian** [iiasa/message_ix#435] [MESSAGE modeling] [answered]
Dear Sir,

I have an ambiguity related to differences between MESSAGE V and MESSAGE-ix.
I have worked with both models. In MESSAGE-V, the model considers multiplying the amount of investment cost of technology and the annual new built capacity of technology. As you remember in MESSAGE V, we enter the investment cost of technology in the MESSAGE model and the model multiplies it by the new capacity. However, in the MESSSAGE-ix I see the model considers the shadow price related to capacity constraint and multiplies it by new capacity.
 In other words, do you follow two different methodologies in these two versions? In MESSAGE-V the user should enter the cost of investment and in MESSAGE-ix the model calculates the dual variable related to the capacity constraint model intead?

Regards

**Comment by zsalimian** [comment]
Dear @volker-krey,
Would you please see the above question?
Regards

**Comment by gorkemgungormetu** [comment]
In my opinion, the structure of MESSAGE-ix seems to be inspired from the PRIMES model which is also based on GAMS solvers. One of the main differences I noticed is the usage of historical capacity as fixed constraint in MESSAGE-V and vintage years in MESSAGE-ix. Also MESSAGE-V excludes the end-of-horizon from the objective function whereas MESSAGE-ix includes. I was comparing the Westeros tutorial files between MESSAGE-ix and MESSAGE-V and it requires some calibration as there are some differences in the equations.

**Comment by behnam-zakeri** [comment, marked as answer]
Dear @zsalimian,
Thanks for asking this and sorry for the delay. The methodologies of the two models are very similar. With respect to your question, if you look into the calculation of cost in MESSAGEix, you see that new capacity `CAP_NEW` is being multiplied by investment cost `inv_cost`:
**https://docs.messageix.org/en/stable/model/MESSAGE/model_core.html#equation-cost-accounting-nodal** 
Marginals (shadow prices in this case) are calculated after the optimization is done and do not have any impact on the results.
